pdns/recursordist/rec-tcp.cc

```diff
@@ -615,6 +622,12 @@ void handleNewTCPQuestion(int fd, FDMultiplexer::funcparam_t&)
     }
 
     bool fromProxyProtocolSource = expectProxyProtocol(addr);
+    const ComboAddress unmappedSource = addr;
+    if (!fromProxyProtocolSource && t_proxyMapping) {
+      if (auto it = t_proxyMapping->lookup(addr)) {
+        addr = it->second;
+      }
+    }
     if (t_allowFrom && !t_allowFrom->match(&addr) && !fromProxyProtocolSource) {
```

Comment:
Not directly related to this PR but the ACL check seems wasteful when `fromProxyProtocolSource` is true?